SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -150,9 +150,18 @@ class PositioningViewController: SitumViewController, GMSMapViewDelegate, UITabl
             }
         }
     }
+    
+    deinit{
+        stopNavigation(status: .canceled)
+        presenter?.removeLocationListener()
+    }
 
     func initializeViewBeforeAppearing(){
-        positionDrawer = GoogleMapsPositionDrawer(mapView: mapView)
+        if positionDrawer == nil {
+            positionDrawer = GoogleMapsPositionDrawer(mapView: mapView)
+        } else {
+            positionDrawer?.makeUserMarkerVisible(visible: false)
+        }
```

Comment:
Aquía igual quedaba máis standard cun `if let positionDrawer = positionDrawer` non?